redbot/core/bot.py

```diff
@@ -1135,10 +1135,16 @@ async def _pre_connect(self) -> None:
                 python_version_changed = True
         else:
             if self._cli_flags.no_cogs is False:
-                packages.extend(await self._config.packages())
+                packages.update(dict.fromkeys(await self._config.packages()))
 
             if self._cli_flags.load_cogs:
-                packages.extend(self._cli_flags.load_cogs)
+                packages.update(dict.fromkeys(self._cli_flags.load_cogs))
+            if self._cli_flags.unload_cogs:
+                for x in self._cli_flags.unload_cogs:
+                    if x not in self._cli_flags.unload_cogs:
+                        self._cli_flags.unload_cogs.remove(x)
+                        continue
+                    del packages[x]
```

Comment:
We don't use the list in `self._cli_flags.unload_cogs` so there is no reason to remove packages from it. Additionally, this wouldn't actually work properly because you're trying to remove an item from a list while iterating over it.
```suggestion
                for package in self._cli_flags.unload_cogs:
                    packages.pop(package, None)
```